src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntity_AlloyBlastSmelter.java

```diff
@@ -255,6 +261,11 @@ public boolean checkRecipe(final ItemStack aStack) {
 							}
 						}
 					}
+					tInputs = Arrays.copyOfRange(tInputList.toArray(new ItemStack[tInputList.size()]), 0, tInputList.size());
+					tRecipe = GTPP_Recipe.GTPP_Recipe_Map.sAlloyBlastSmelterRecipes.findRecipe(this.getBaseMetaTileEntity(), false, gregtech.api.enums.GT_Values.V[tTier], tFluids, tInputs);
+					if ((tRecipe != null) && (tRecipe.isRecipeInputEqual(false, tFluids, tInputs))) {
```

Comment:
why you check the recipe again when it already find one?

Reply:
This was how the original code did it. I just moved the lines to the input bus checks.

Reply:
alk's crap code probably, could you just remove them?